dipy/testing/tests/test_decorators.py

```diff
@@ -51,3 +51,36 @@ def f():
     del HAVE_AMODULE
     f.__doc__ = docstring
     assert_raises(NameError, doctest_skip_parser, f)
+
+def test_warn_on_positional_kwonly():
+    @warn_on_positional_args(valid="")
+    def func_with_kwonly_args(a, b, *, c=3):
+        return a + b + c
+
+    @warn_on_positional_args(valid="")
+    def func_with_default_args(a=1, b=2, c=3):
+        return a + b + c
+
+    # Test case: No warning expected when keyword-only argument is used correctly
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        func_with_kwonly_args(a=1, b=2, c=3)
+        assert len(w) == 0, "Expected no warnings"
+
+    # Test case: No warning expected when all arguments have default values
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        func_with_default_args(a=1, b=2, c=3)
+        assert len(w) == 0, "Expected no warnings"
+
+    # Test case: Warning expected when keyword-only argument is used positionally
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        func_with_kwonly_args(1, 2, 3)
```

Comment:
Test does not pass because a `TypeError` is raised. Catch it, or adapt the function as suggested in prior comment.